src/pykeen/models/nbase.py

```diff
@@ -429,92 +342,123 @@ def append_weight_regularizer(
 
     def forward(
         self,
-        h_indices: Optional[torch.LongTensor],
-        r_indices: Optional[torch.LongTensor],
-        t_indices: Optional[torch.LongTensor],
+        h_indices: torch.LongTensor,
+        r_indices: torch.LongTensor,
+        t_indices: torch.LongTensor,
         slice_size: Optional[int] = None,
-        slice_dim: Optional[str] = None,
+        slice_dim: int = 0,
     ) -> torch.FloatTensor:
         """Forward pass.
 
-        This method takes head, relation and tail indices and calculates the corresponding score.
-
-        All indices which are not None, have to be either 1-element or have the same shape, which is the batch size.
+        This method takes head, relation and tail indices and calculates the corresponding scores.
+        It supports broadcasting.
 
         :param h_indices:
-            The head indices. None indicates to use all.
+            The head indices.
         :param r_indices:
-            The relation indices. None indicates to use all.
+            The relation indices.
         :param t_indices:
-            The tail indices. None indicates to use all.
+            The tail indices.
+
+        :return:
+            The scores
+        """
+        h, r, t = self._get_representations(h=h_indices, r=r_indices, t=t_indices)
+        return self.interaction.score(h=h, r=r, t=t, slice_size=slice_size, slice_dim=slice_dim)
+
+    def score_hrt(self, hrt_batch: torch.LongTensor) -> torch.FloatTensor:
+        """Forward pass.
+
+        This method takes head, relation and tail of each triple and calculates the corresponding score.
+
+        :param hrt_batch: shape: (batch_size, 3), dtype: long
+            The indices of (head, relation, tail) triples.
+
+        :return: shape: (batch_size, 1), dtype: float
+            The score for each triple.
+        """
+        # Note: slicing cannot be used here: the indices for score_hrt only have a batch
+        # dimension, and slicing along this dimension is already considered by sub-batching.
+        h, r, t = self._get_representations(h=hrt_batch[:, 0], r=hrt_batch[:, 1], t=hrt_batch[:, 2])
```

Comment:
just defer to foward pass? or just `self()`?

Reply:
no, because this is a fast path skipping some of the more involved logic.